Sources/Operator/graph_operator.hpp

```diff
@@ -109,7 +108,7 @@ class GraphOperator : public AbstractOperator {
     auto lop = lhs.operator_;
     auto rop = rhs.operator_;
 
-    return GraphOperator(lhs.GetHilbert(), lop + rop);
+    return GraphOperator(lhs.GetHilbert().shared_from_this(), lop + rop);
```

Comment:
is this shared_from_this really necessary? Isn't it enough to have followed 1. in https://pybind11.readthedocs.io/en/stable/advanced/smart_ptrs.html ?

Reply:
This is possibly even uglier than calling `->` everywhere !

Reply:
It is only necessary in places that actually retain an owning reference to the Hilbert object (as the new GraphOperator does here).

When using the Hilbert space inside the operator methods it is not needed. (The alternative would be returning a `shared_ptr` all the time, but that introduces unnecessary overhead when `GetHilbert` is called.